server/src/main/java/org/apache/iotdb/db/query/executor/groupby/SlidingWindowGroupByExecutorFactory.java

```diff
@@ -20,16 +20,51 @@
 package org.apache.iotdb.db.query.executor.groupby;
 
 import org.apache.iotdb.db.qp.constant.SQLConstant;
+import org.apache.iotdb.db.query.aggregation.AggregateResult;
 import org.apache.iotdb.db.query.executor.groupby.impl.EmptyQueueSlidingWindowGroupByExecutor;
 import org.apache.iotdb.db.query.executor.groupby.impl.MonotonicQueueSlidingWindowGroupByExecutor;
 import org.apache.iotdb.db.query.executor.groupby.impl.NormalQueueSlidingWindowGroupByExecutor;
 import org.apache.iotdb.db.query.executor.groupby.impl.SmoothQueueSlidingWindowGroupByExecutor;
-import org.apache.iotdb.db.utils.TypeInferenceUtils;
-import org.apache.iotdb.tsfile.exception.write.UnSupportedDataTypeException;
 import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
 
+import java.util.Comparator;
+
 public class SlidingWindowGroupByExecutorFactory {
 
+  /** comparators used for MonotonicQueueSlidingWindowGroupByExecutor */
+
+  // return a value greater than 0 if o1 is numerically greater than o2
+  private static final Comparator<AggregateResult>[] maxComparators =
```

Comment:
It's better to use `Map<TSDataType, Comparator>` than `ordinal()` indexed because it's an implict restrict that we can't change the order of `TSDataType` definition. If someone changed it,  some bugs will be out.

Reply:
fixed~